pinot-plugins/pinot-minion-tasks/pinot-minion-builtin-tasks/src/main/java/org/apache/pinot/plugin/minion/tasks/mergerollup/MergeRollupTaskGenerator.java

```diff
@@ -330,14 +354,18 @@ public List<PinotTaskConfig> generateTasks(List<TableConfig> tableConfigs) {
               // TODO: If there are many small merged segments, we should merge them again
               selectedSegmentsForBucket = new ArrayList<>();
               hasUnmergedSegments = false;
+              isAllSegmentsReadyToMerge = true;
               bucketStartMs = (startTimeMs / bucketMs) * bucketMs;
               bucketEndMs = bucketStartMs + bucketMs;
-              if (!isValidBucketEndTime(bucketEndMs, bufferMs, lowerMergeLevel, mergeRollupTaskMetadata)) {
+              if (!isValidBucketEndTime(bucketEndMs, bufferMs, lowerMergeLevel, mergeRollupTaskMetadata, isLookBack)) {
                 break;
               }
-              if (!isMergedSegment(preSelectedSegment, mergeLevel)) {
+              if (!isMergedSegment(preSelectedSegment, mergeLevel, sortedMergeLevels)) {
```

Comment:
code pattern occurs in both sides of if/else: can we factor it out?

Reply:
More specifically: line 322 seems to have exact same code as here with same action:
`if (!isMergedSegment(preSelectedSegment, mergeLevel, sortedMergeLevels)) {
        hasUnmergedSegments = true;
}`

Can we factor that at start of if that is in line 330:

`if (endTimeMs >= bucketStartMs) {`

Reply:
Actually, this function is not called on both sides of if/else. Github folded L351-L356, and that may lead to the confusion?